### PR TITLE
Camel 17600 add missing azure features

### PR DESCRIPTION
**Pull request by jojansen**
add missing features to connec to for azure-servicebus:

- azure-servicebus
- azure-identity

Depending on new servicemix bundle-releases.

### CODE REVIEW

File: platforms/karaf/features/src/main/resources/features.xml [hunk 1]
```diff
@@ -364,37 +364,78 @@
...
+    <feature version='${project.version}' prerequisite='true'>camel-azure-core</feature>
+    <bundle dependency='true'>mvn:org.apache.qpid/qpid-jms-client/${qpid-jms-client-version}</bundle>
+    <bundle dependency='true'>mvn:org.apache.qpid/proton-j/${qpid-proton-j-version}</bundle>
+    <feature version='${project.version}'>camel-azure-storage-blob</feature>
+    <bundle dependency='true'>mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.azure-core-amqp/${azure-core-amqp-bundle-version}</bundle>
+    <bundle dependency='true'>mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.azure-messaging-eventhubs/${azure-messaging-eventhubs-bundle-version}</bundle>
+    <bundle dependency='true'>mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.azure-messaging-eventhubs-checkpointstore-blob/${azure-messaging-eventhubs-checkpointstore-blob-bundle-version}</bundle>
+    <bundle>mvn:org.apache.camel/camel-azure-eventhubs/${project.version}</bundle>
+  </feature>
+  <feature name='camel-azure-identity' version='${camel.version}'  description='Feature to use the Azure-Identity functionality.'>
```
Comment (oscerd) [inline platforms/karaf/features/src/main/resources/features.xml:393]: There is no component named camel-azure-identity in Camel.

File: platforms/karaf/features/src/main/resources/features.xml [hunk 2]
```diff
@@ -364,37 +364,78 @@
...
+  <feature name='camel-azure-core' version='${camel.version}' description='Core bundles for Azure usage.' start-level='50' hidden='true'>
+    <feature version='${camel.version}' dependency='true'>camel-core</feature>
+
     <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-core/${jackson2-version}</bundle>
     <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-databind/${jackson2-version}</bundle>
     <bundle dependency='true'>mvn:com.fasterxml.jackson.core/jackson-annotations/${jackson2-version}</bundle>
     <bundle dependency='true'>mvn:com.fasterxml.jackson.dataformat/jackson-dataformat-xml/${jackson2-version}</bundle>
     <bundle dependency='true'>mvn:com.fasterxml.jackson.module/jackson-module-jaxb-annotations/${jackson2-version}</bundle>
     <bundle dependency='true'>mvn:org.codehaus.woodstox/stax2-api/${stax2v4-api-bundle-version}</bundle>
-    <bundle dependency='true'>mvn:com.fasterxml.jackson.datatype/jackson-datatype-jsr310/${jackson2-version}</bundle>
```
Comment (oscerd) [inline platforms/karaf/features/src/main/resources/features.xml:375]: Why? There is no jackson.version in parent/pom from camel-dependencies.

**Review by oscerd — CHANGES_REQUESTED** [review]
I think you need to review a bit, you're introducing new modules and currently is really hard to track. If you need to define a feature like "azure-core" name it "azure-core" and not "camel-azure-core". This will avoid confusion.

Also I would avoid having superset feature like azure-core and azure-identity, because they will become hard to maintain in a short time. Better to have a single component feature will all inside. If I want to update just one client library, I don't have to deal with the alignments for all the others.

For me, in this form, it's -1.

Thanks anyway.

**Review by davsclaus — CHANGES_REQUESTED** [review]
As Andrea says

**Pull request closed without merging**